src/hotspot/share/gc/parallel/objectStartArray.inline.hpp

```diff
@@ -45,21 +45,14 @@ HeapWord* ObjectStartArray::object_start(HeapWord* const addr) const {
 
 HeapWord* ObjectStartArray::block_start_reaching_into_card(HeapWord* const addr) const {
   const uint8_t* entry = entry_for_addr(addr);
-
-  uint8_t offset;
-  while (true) {
-    offset = *entry;
-
-    if (offset < CardTable::card_size_in_words()) {
-      break;
-    }
-
+  uint8_t offset = *entry;
```

Comment:
I kind of like the original style, as it has a single de-reference. (It's indeed rather subjective.) Also, this change is completely unrelated to this ticket.

Reply:
Reverted.

Reply:
Should I revert the `SerialBlockOffsetTable::block_start_reaching_into_card` too?